src/Features/CSharp/Portable/UseLocalFunction/CSharpUseLocalFunctionCodeFixProvider.cs

```diff
@@ -46,112 +46,124 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
+                var delegateType = (INamedTypeSymbol)semanticModel.GetTypeInfo(anonymousFunction, cancellationToken).ConvertedType;
+                var parameterList = GenerateParameterList(anonymousFunction, delegateType.DelegateInvokeMethod);
 
-                var currentLocalDeclaration = currentRoot.GetCurrentNode(originalLocalDeclaration);
-                var currentLambda = currentRoot.GetCurrentNode(originalLambda);
+                var currentLocalDeclaration = currentRoot.GetCurrentNode(localDeclaration);
+                var currentAnonymousFunction = currentRoot.GetCurrentNode(anonymousFunction);
 
                 currentRoot = ReplaceAnonymousWithLocalFunction(
                     document.Project.Solution.Workspace, currentRoot,
-                    currentLocalDeclaration, currentLambda,
-                    delegateType, parameterList, explicitInvokeCalls.Select(node => currentRoot.GetCurrentNode(node)).ToImmutableArray(),
-                    cancellationToken);
+                    currentLocalDeclaration, currentAnonymousFunction,
+                    delegateType.DelegateInvokeMethod, parameterList);
+
+                // these invocations might actually be inside the local function! so we have to do this separately
+                currentRoot = ReplaceInvocations(
+                    document.Project.Solution.Workspace, currentRoot,
+                    delegateType.DelegateInvokeMethod, parameterList,
+                    invocations.Select(node => currentRoot.GetCurrentNode(node)).ToImmutableArray());
             }
 
             editor.ReplaceNode(root, currentRoot);
         }
 
-        private SyntaxNode ReplaceAnonymousWithLocalFunction(
+        private static SyntaxNode ReplaceAnonymousWithLocalFunction(
             Workspace workspace, SyntaxNode currentRoot,
-            LocalDeclarationStatementSyntax localDeclaration, LambdaExpressionSyntax lambda,
-            INamedTypeSymbol delegateType, ParameterListSyntax parameterList,
-            ImmutableArray<MemberAccessExpressionSyntax> explicitInvokeCalls,
-            CancellationToken cancellationToken)
+            LocalDeclarationStatementSyntax localDeclaration, AnonymousFunctionExpressionSyntax anonymousFunction,
+            IMethodSymbol delegateMethod, ParameterListSyntax parameterList)
         {
-            var newLocalFunctionStatement = CreateLocalFunctionStatement(
-                localDeclaration, lambda, delegateType, parameterList, cancellationToken);
-
-            newLocalFunctionStatement = newLocalFunctionStatement.WithTriviaFrom(localDeclaration)
-                                                                 .WithAdditionalAnnotations(Formatter.Annotation);
+            var newLocalFunctionStatement = CreateLocalFunctionStatement(localDeclaration, anonymousFunction, delegateMethod, parameterList)
+                .WithTriviaFrom(localDeclaration)
```

Comment:
>WithTriviaFrom [](start = 17, length = 14)

This makes me think it would be good to tests with trivia.

For instance:
```C#
/* comment 1 */
Action [||]local = null; // comment 2
/* comment 3 */
local = () => local.Invoke(); // comment 4
```

I'm not sure how to handle clashes like this (between comments 2 and 4 in this example).

Reply:
I am sorry but I would prefer not going out of my way to solve a design problem with comments here. This PR already fixes 6 different bugs, all of which were unrelated to the issue I was originally trying to solve.

Reply:
Thanks